.github/workflows/osx.yml

```diff
@@ -55,10 +55,18 @@ jobs:
         if: success() && ! contains(matrix.type, 'cmake')
         run: ./mac_build/buildMacBOINC-CI.sh --no_shared_headers
 
+      - name: test universal
```

Comment:
I don't know why this test is outside the build procedure. For me, if build script has failed to build any binary in an expected mode, it should immediately fail, so I'd suggest to put this verification to `./mac_build/buildMacBOINC-CI.sh` script and run on every built target (and not just by enumerating files in the directory). Then it will be more consistent and clear.

Reply:
I preferer to do external check and not inside the scrip because if we change the check, it invalidate the cache and take more time to build. Also it I am doing this check after all the build is done, to change the script for every target I will insert some bugs, ratter to change your script that already working, and we need to verify it again. It will take more time.
Also we have many lines of output in the test, we need some other part to see it very quickly what happen. It  more readability.

Reply:
I still would like to have this inside the ./mac_build/buildMacBOINC-CI.sh script because in this case it will be more logical: we build binary and check it right after the build. Now your solution have decoupled binary verification and it's hard to remember to fix verification list in case of adding/removing binaries from the build.
So I strongly recommend to follow my suggestion and move this check to the ./mac_build/buildMacBOINC-CI.sh file.